src/native-common/View.tsx

```diff
@@ -9,29 +9,202 @@
+    touchableHandleResponderMove: (e: React.SyntheticEvent<any>) => void;
+    touchableHandleResponderRelease: (e: React.SyntheticEvent<any>) => void;
+    touchableHandleResponderTerminate: (e: React.SyntheticEvent<any>) => void;
+
+    private _mixinIsApplied = false;
+    private _childrenKeys: ChildKey[];
+
     constructor(props: Types.ViewProps) {
         super(props);
-
+        this._updateMixin(props, true);
         this._buildInternalProps(props);
     }
 
     componentWillReceiveProps(nextProps: Types.ViewProps) {
+        this._updateMixin(nextProps, false);
         this._buildInternalProps(nextProps);
     }
 
+    componentWillUpdate(nextProps: Types.ViewProps, nextState: {}) {
+        //
+        // Exit fast if not an "animated children" case
+        if (!(nextProps.animateChildEnter || nextProps.animateChildMove || nextProps.animateChildLeave)) {
+            return;
+        }
+
+        // Each time the component receives new children, animates insertions, removals,
+        // and moves that occurred since the previous render. Uses React Native's
+        // LayoutAnimation API to achieve this.
+        //
+        // Caveats:
+        //   - The animations are not scoped. All layout changes in the app that occur during the
+        //     next bridge transaction will be animated. This is due to a limitation in React Native's
+        //     LayoutAnimation API.
+        //   - Removals are not animated. The removed item disappears instantly. Items whose positions
+        //     were affected by the removal are animated into their new positions. This is due to a
+        //     limitation in React Native's LayoutAnimation API.
+        //
+
+        // The web implementation doesn't support string refs. For consistency, do the same assert
+        // in the native implementation.
+        assert(
+            findInvalidRefs(this.props.children).length === 0,
+            'Invalid ref(s): ' + JSON.stringify(findInvalidRefs(this.props.children)) +
+            ' Only callback refs are supported when using child animations on a `View`'
+        );
+
+        let prevChildrenKeys = this._childrenKeys;
+        let nextChildrenKeys = extractChildrenKeys(nextProps.children);
+        this._childrenKeys = nextChildrenKeys;
+        if (_childrenEdited(prevChildrenKeys, nextChildrenKeys)) {
+            let updateConfig: RN.IUpdateLayoutAnimationConfig = {
+                delay: 0,
+                duration: 300,
+                type: LayoutAnimation.Types.easeOut
+            };
+            let createConfig: RN.ICreateLayoutAnimationConfig = {
+                delay: 75,
+                duration: 150,
+                type: LayoutAnimation.Types.linear,
+                property: LayoutAnimation.Properties.opacity
+            };
+            var configDictionary: RN.ILayoutAnimationConfig = {
+                duration: 300,
+            };
+
+            if (this.props.animateChildMove) {
```

Comment:
Hmm, something is wrong here, even this should be "nextprops", I think I need to update. Thinking.